tests/sanity.py

```diff
@@ -0,0 +1,194 @@
+########
+# Copyright (c) 2015 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+import json
+import logging
+import os
+
+import boto.ec2
+from retrying import retry
+from fabric.api import settings as fabric_settings
+from fabric.api import run as fabric_run
+
+
+REGION = 'us-east-1'
+CENTOS_7_AMI_ID = 'ami-96a818fe'
+USER = 'centos'
+INSTANCE_TYPE = 'm3.medium'
+RESOURCE_NAME = 'cloudify-bootstrap-sanity-travis'
+
+logging.basicConfig(level=logging.INFO,
+                    format='%(asctime)s %(levelname)s %(message)s')
+lgr = logging.getLogger('sanity')
+lgr.setLevel(logging.INFO)
+
+aws_access_key = os.environ['AWS_ACCESS_KEY']
+aws_secret_access_key = os.environ['AWS_SECRET_ACCESS_KEY']
+ssh_public_key = os.environ['SSH_PUBLIC_KEY']
+ssh_private_key = os.environ['SSH_PRIVATE_KEY']
+
+
+def delete_security_group(conn):
+    sgs = [
```

Comment:
regarding all the `get_all_XXXX`, is there any server side filter to avoid filtering on the client?
there could be many keypairs/security_groups/instances.